pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/LoaderUtils.java

```diff
@@ -90,4 +97,28 @@ public static void writeIndexToV3Format(SegmentDirectory.Writer segmentWriter, S
     }
     return stringList;
   }
+
+  /**
+   * Try to recover a segment from reload failure, no effect for normal segments.
```

Comment:
expand this comment outlining the scenarios that can happen, and describe why we are recovering this way. Give a pointer to the reload() method and explain what the failure (i.e. server restart) points can be, and what cruft is left behind at each stage.